src/components/UserProfile/Integration.js

```diff
@@ -52,12 +58,19 @@ const trackFenceAction = async ({ fence, fenceDetails, category, action, label }
   await trackUserInteraction({ category, action, label });
 };
 
-const disconnect = async ({ fence, api, setConnecting, effects, setError }) => {
+const disconnect = async ({
+  fence,
+  api,
+  setConnecting,
+  setError,
+  removeFenceConnection,
+  removeFenceStudies,
+}) => {
   setConnecting(true);
   try {
     await deleteFenceTokens(api, fence);
-    await effects.setIntegrationToken(fence, null);
-    await effects.removeFenceConnection(fence);
+    removeFenceConnection(fence);
+    removeFenceStudies(fence);
     await trackFenceAction({
       fence,
       fenceDetails: '',
```

Comment:
try with no `catch` not considered a good pratice. I know it's not from that PR, but worth looking into

Reply:
There is a `catch` clause. It is just not visible in the snippet above.